server/connector.go

```diff
@@ -189,6 +191,26 @@ func (c *connectorImpl) findAndConnectBackend(ctx context.Context, frontendConn
 	c.metrics.ActiveConnections.Add(1)
 	defer c.metrics.ActiveConnections.Add(-1)
 
+	// PROXY protocol implementation
+
+	remoteAddr, err := net.ResolveIPAddr("ip", strings.Split(backendHostPort, ":")[0])
```

Comment:
You can actually use https://pkg.go.dev/net@go1.17.3#SplitHostPort to split that apart and grab the hostname, such as

```suggestion
      remoteHost, _, _ := net.SplitHostPort(backendHostPort)
	remoteAddr, err := net.ResolveIPAddr("ip", remoteHost)
```